src/bitmessagekivy/tests/telenium_process.py

```diff
@@ -0,0 +1,80 @@
+import os
+import shutil
+import tempfile
+from time import time, sleep
+
+from telenium.tests import TeleniumTestCase
+
+_files = (
+    'keys.dat', 'debug.log', 'messages.dat', 'knownnodes.dat',
+    '.api_started', 'unittest.lock'
+)
+
+tmp_db_file = (
+    'keys.dat', 'messages.dat'
+)
+
+
+def cleanup(home=None, files=_files):
+    """Cleanup application files"""
+    if not home:
+        home = tempfile.gettempdir()
+    for pfile in files:
+        try:
+            os.remove(os.path.join(home, pfile))
+        except OSError:
+            pass
+
+
+def set_temp_data():
+    """Set temp data in tmp directory"""
+    for file_name in tmp_db_file:
+        old_source_file = os.path.join(
+            os.path.abspath(os.path.dirname(__file__)), 'sampleData', file_name)
+        new_destination_file = os.path.join(os.environ['BITMESSAGE_HOME'], file_name)
+        shutil.copyfile(old_source_file, new_destination_file)
+
+
+class TeleniumTestProcess(TeleniumTestCase):
+    """Setting Screen Functionality Testing"""
+    cmd_entrypoint = [os.path.join(os.path.abspath(os.getcwd()), 'src', 'mock', 'main.py')]
+
+    @classmethod
+    def setUpClass(cls):
+        """Setupclass is for setting temp environment"""
+        os.environ["BITMESSAGE_HOME"] = tempfile.gettempdir()
+        set_temp_data()
+        super(TeleniumTestProcess, cls).setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Ensures that pybitmessage stopped and removes files"""
+        # pylint: disable=no-member
+        cleanup()
+        try:
+            cls.cli.app_quit()
+        except:
+            pass
+
+        try:
+            cls.process.kill()
+        except:
+            pass
+
+    def drag(self, xpath1, xpath2):
+        """this method is for dragging"""
+        self.cli.drag(xpath1, xpath2, 1.5)
+        self.cli.sleep(0.3)
+
+    def assertCheckScrollDown(self, scroll_distance, destination, timeout=-1):
+        """this method is for checking scroll"""
+        start = time()
+        while True:
+            if scroll_distance <= destination:
```

Comment:
It looks like here `scroll_distance` is a value, rather than a reference, which make the test less useful.

Reply:
Yes I know scroll_distance is the value of the y-axis which starts from 0.0-1.0 so I am checking if scroll values are less than or equal to 0.0 then it means that someone has scrolled the nav drawer if the condition is satisfied then the test passes otherwise it given and fail result

Reply:
I think what you said is correct, the problem is that you retrieve the value only once, before calling this method. In other words, the loop doesn't check if the value has changed. The loop is superfluous as it is repeating exactly the same thing. You should pass a reference, e.g. a function, or a telenium selector, not a value.

Reply:
In other words, `scroll_distance` is a float and the value is assigned before this method is called. It doesn't change while the loop is running.

Reply:
ok